### PR TITLE
Add document on HTTP/2

### PR DESCRIPTION
**Pull request by bjorncs**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Review by jobergum — APPROVED** [review]

**Comment by jobergum** [comment]
Nice!

File: en/performance/http2.html [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+</p>
+<p>
+    Vespa provides a Java implementation of HTTP/2 + Document v1 in <a href="/en/vespa-feed-client.html">vespa-feed-client</a>.
+    This client is available both as a Java API and a command line client.
+</p>
+
+<h2>Performance tuning</h2>
+<h3>Client</h3>
+<p>
+    The number of multiple concurrent requests per connection is typically adjustable on HTTP/2 clients/libraries.
```
Comment (kkraune) [inline en/performance/http2.html:43]:
```suggestion
    The number of multiple concurrent requests per connection is typically adjustable in HTTP/2 clients/libraries.
```

File: en/performance/http2.html [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    Document v1 API is designed for high concurrency and can easily handle several hundred concurrent requests.
+    Its implementation is asynchronous and max concurrency is not restricted directly by any thread pool size
+    - you may configure your client to use several hundred concurrent streams per connection for this API.
+    Other APIs such as the <a href="/en/query-api.html">Query API</a> is backed by a synchronous implementation,
+    and max concurrency is restricted by the <a href="/en/performance/container-tuning.html#container-worker-threads">underlying thread pool size</a>.
+    Too many concurrent streams may result in the container rejecting requests with 503 responses.
+</p>
+<p>
+    There are still good arguments for using multiple TCP connections - even with HTTP/2.
```
Comment (kkraune) [inline en/performance/http2.html:52]:
```suggestion
    There are still good arguments for using multiple TCP connections - even with HTTP/2:
```

File: en/performance/index.html [hunk 3]
```diff
@@ -31,6 +31,7 @@ <h2>Benchmarking and tuning</h2>
   <li><a href="container-http.html">Container Http performance testing using Gatling</a></li>
   <li><a href="container-tuning.html">Container tuning</a>: JVM, container, docproc</li>
   <li><a href="../reference/vespa-cmdline-tools.html#vespa-fbench">vespa-fbench</a>: Reference documentation</li>
+  <li><a href="container-tuning.html">HTTP/2</a>: improve HTTP performance using HTTP/2</li>
```
Comment (kkraune) [inline en/performance/index.html:34]: wrong link, should link to http2.html ?

**Review by kkraune — APPROVED** [review]
Great doc!

File: en/performance/http2.html [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    Both HTTP/1.1 and HTTP/2 will be served over the same connector using the <a href="https://datatracker.ietf.org/doc/html/rfc7301">TLS ALPN Extension</a>.
+    The Application-Layer Protocol Negotiation (ALPN) extension allows the client to send a list of supported protocols during TLS handshake.
+    The container selects a supported protocol from that list.
+</p>
+<p>
+    The <a href="https://datatracker.ietf.org/doc/html/rfc7540">HTTP/2 specification</a> dictates multiple requirements for TLS connection.
+    Vespa may enforce some or all of these restrictions. See the HTTP/2 specification for the full list. The most significant are listed below:
+</p>
+<ul>
+    <li>Client must use least TLSv1.2.</li>
```
Comment (jonmv) [inline en/performance/http2.html:22]:
```suggestion
    <li>Client must use at least TLSv1.2.</li>
```

File: en/performance/http2.html [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+<h2>Enabling HTTP/2 on container</h2>
+<p>
+    HTTP/2 is enabled on a container for all connectors with TLS/HTTPS configured (requires Vespa 7.418.23 or newer).
+    Both HTTP/1.1 and HTTP/2 will be served over the same connector using the <a href="https://datatracker.ietf.org/doc/html/rfc7301">TLS ALPN Extension</a>.
+    The Application-Layer Protocol Negotiation (ALPN) extension allows the client to send a list of supported protocols during TLS handshake.
+    The container selects a supported protocol from that list.
+</p>
+<p>
+    The <a href="https://datatracker.ietf.org/doc/html/rfc7540">HTTP/2 specification</a> dictates multiple requirements for TLS connection.
```
Comment (jonmv) [inline en/performance/http2.html:18]:
```suggestion
    The <a href="https://datatracker.ietf.org/doc/html/rfc7540">HTTP/2 specification</a> dictates multiple requirements for the TLS connection.
```

File: en/performance/http2.html [hunk 6]
```diff
@@ -0,0 +1,80 @@
+---
+# Copyright Verizon Media. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+title: "HTTP/2"
+---
+<p>
+    This document contains HTTP/2 performance considerations on the container -
+    see <a href="container-tuning.html">Container tuning</a> for general tuning of container clusters.
```
Comment (jonmv) [inline en/performance/http2.html:7]:
```suggestion
    This document contains HTTP/2 performance considerations on the container—see
    <a href="container-tuning.html">Container tuning</a> for general tuning of container clusters.
```

File: en/performance/http2.html [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+    The Application-Layer Protocol Negotiation (ALPN) extension allows the client to send a list of supported protocols during TLS handshake.
+    The container selects a supported protocol from that list.
+</p>
+<p>
+    The <a href="https://datatracker.ietf.org/doc/html/rfc7540">HTTP/2 specification</a> dictates multiple requirements for TLS connection.
+    Vespa may enforce some or all of these restrictions. See the HTTP/2 specification for the full list. The most significant are listed below:
+</p>
+<ul>
+    <li>Client must use least TLSv1.2.</li>
+    <li>Client must provided target domain in TLS Server Name Indication (SNI) Extension.</li>
```
Comment (jonmv) [inline en/performance/http2.html:23]:
```suggestion
    <li>Client must provide target domain with the TLS Server Name Indication (SNI) Extension.</li>
```

File: en/performance/http2.html [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+    <li>Client must not use any of the banned <a href="https://datatracker.ietf.org/doc/html/rfc7540#appendix-A">TLSv1.2 ciphers</a>.</li>
+</ul>
+
+<h2>Feeding over HTTP/2</h2>
+<p>
+    One of the major improvements with HTTP/2 is multiplexing of multiple concurrent requests over a single TCP connections.
```
Comment (jonmv) [inline en/performance/http2.html:29]:
```suggestion
    One of the major improvements with HTTP/2 is multiplexing of multiple concurrent requests over a single TCP connection.
```

File: en/performance/http2.html [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+    This allows for high-throughput feeding over the <a href="/en/reference/document-v1-api-reference.html">Document v1 HTTP API</a>
+    without the overhead with hundreds of parallel connections that HTTP/1.1 would incur.
+    HTTP/2 + Document v1 is a great alternative to <a href="/en/vespa-http-client.html">Vespa HTTP Client</a>
+    which does not require a JDK 8+ environment for the client to live in.
```
Comment (jonmv) [inline en/performance/http2.html:33]:
```suggestion
    This allows for high-throughput feeding through the <a href="/en/reference/document-v1-api-reference.html">Document v1 HTTP API</a>,
    with a simple one-operation–one-request model, but without the overhead of 
    hundreds of parallel connections that HTTP/1.1 would require for sufficient concurrency.
    HTTP/2 + Document v1 is a great alternative to the <a href="/en/vespa-http-client.html">Vespa HTTP Client</a>,
    which implements a custom protocol, and requires a JDK 8+ client environment. 
```

File: en/performance/http2.html [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+    This client is available both as a Java API and a command line client.
+</p>
+
+<h2>Performance tuning</h2>
+<h3>Client</h3>
+<p>
+    The number of multiple concurrent requests per connection is typically adjustable on HTTP/2 clients/libraries.
+    Document v1 API is designed for high concurrency and can easily handle several hundred concurrent requests.
+    Its implementation is asynchronous and max concurrency is not restricted directly by any thread pool size
+    - you may configure your client to use several hundred concurrent streams per connection for this API.
```
Comment (jonmv) [inline en/performance/http2.html:46]:
```suggestion
    Its implementation is asynchronous and max concurrency is not restricted by a thread pool size, 
    so configure your client to allow enough concurrent requests/streams to saturate the feed container.
```

File: en/performance/http2.html (hunk 2 above)
Comment (jonmv) [reply en/performance/http2.html:52]:
```suggestion
    There are also still some reasons to use multiple TCP connections—even with HTTP/2:
```

File: en/performance/http2.html [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+</p>
+<p>
+    There are still good arguments for using multiple TCP connections - even with HTTP/2.
+</p>
+<ul>
+    <li><strong>Utilize multiple containers</strong>.
+        A single container may not saturate the content layer.
+        A client may have to use more connections than container nodes if the containers are behind a load balancer.</li>
+    <li><strong>Higher throughput</strong>.
+        Many clients allow for a single thread to operation on a single connection.
```
Comment (jonmv) [inline en/performance/http2.html:59]:
```suggestion
        Many clients allow only for a single thread to operate each connection.
```

File: en/performance/index.html (hunk 3 above)
Comment (jonmv) [reply en/performance/index.html:34]:
```suggestion
  <li><a href="http2.html">HTTP/2</a>: improve HTTP performance using HTTP/2</li>
```

**Comment by jonmv** [comment]
Nice! Suggested some minor changes.

**Comment by bjorncs** [comment]
Thanks for feedback @kkraune

**Pull request merged**